apps/web/lib/actions/create-dots-transfer.ts

```diff
@@ -0,0 +1,45 @@
+"use server";
+
+import { prisma } from "@/lib/prisma";
+import { z } from "zod";
+import { createOrgTransfer } from "../dots/create-org-transfer";
+import { createTransfer } from "../dots/create-transfer";
+import { authActionClient } from "./safe-action";
+
+const schema = z.object({
+  workspaceId: z.string(),
+  dotsUserId: z.string(),
+  payoutId: z.string(),
+  amount: z.number(),
+  fee: z.number(),
+});
+
+export const createDotsTransferAction = authActionClient
+  .schema(schema)
+  .action(async ({ parsedInput, ctx }) => {
+    const { workspace } = ctx;
+    const { dotsUserId, payoutId, amount, fee } = parsedInput;
+
+    if (!workspace.dotsAppId) {
+      throw new Error("Dots app not found");
+    }
+
+    const [transfer, orgTransfer] = await Promise.all([
+      createTransfer({
+        amount,
+        dotsAppId: workspace.dotsAppId,
+        dotsUserId,
+      }),
+      createOrgTransfer({
+        amount: fee,
+        dotsAppId: workspace.dotsAppId,
+      }),
```

Comment:
@steven-tey we may want to fetch this fee on the back-end rather than letting the front-end pass any amount

Reply:
great idea, will restrict this to only pass `payoutId` and fetch the details server side